socorro/external/elasticsearch/index_cleaner.py

```diff
@@ -0,0 +1,82 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import datetime
+import pyelasticsearch
+import re
+
+from configman import Namespace, RequiredConfig
+from configman.converters import class_converter
+
+from socorro.lib.datetimeutil import utc_now
+
+
+class IndexCleaner(RequiredConfig):
+    """Delete old Elasticsearch indices from our databases. """
+
+    required_config = Namespace()
+    required_config.add_option(
+        'retention_policy',
+        default=26,
+        doc='Number of weeks to keep an index alive. ',
+    )
+    required_config.namespace('elasticsearch')
+    required_config.elasticsearch.add_option(
+        'elasticsearch_class',
+        default='socorro.external.elasticsearch.connection_context.'
+                'ConnectionContext',
+        from_string_converter=class_converter,
+        reference_value_from='resource.elasticsearch',
+    )
+    required_config.elasticsearch.add_option(
+        'elasticsearch_index_regex',
+        default='socorro[0-9]{6}$',
+        reference_value_from='resource.elasticsearch',
+    )
+
+    def __init__(self, config):
+        super(IndexCleaner, self).__init__()
+        self.config = config
+
+    def delete_old_indices(self):
+        now = utc_now()
+        policy_delay = datetime.timedelta(weeks=self.config.retention_policy)
+        time_limit = (now - policy_delay).replace(tzinfo=None)
+
+        es = pyelasticsearch.ElasticSearch(
+            self.config.elasticsearch.elasticsearch_urls,
+            timeout=self.config.elasticsearch.elasticsearch_timeout
+        )
+
+        state = es.cluster_state()
+        indices = state['metadata']['indices'].keys()
+
+        aliases = es.aliases()
+
+        for index in indices:
+            # Some indices look like 'socorro%Y%W_%Y%M%d', but they are
```

Comment:
Those weirdly named indices are created when we reindex data. It has not happened in a while, but it might happen again. That makes me think that we should come up with a better strategy for reindexing and aliasing. We discussed that a bit last year but never took time to actually do anything about it.